### PR TITLE
Image Gallery with captions

### PR DESCRIPTION
**Pull request by col-tom**
**This includes**

* Simple image gallery with captions-editing in the backend UI
* Preview image with the first image that acts as a _button_ to open full gallery
* `watch mode` to have a better DX (simple one, fast refresh not there yet)
* Updated deps

This is the preparation to get `copyright` and `caption` from the media assets in the gutenberg blocks available.

### CODE REVIEW

File: src/image_gallery/index.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    const hasImages = attributes.images && !!attributes.images.length;
+    const onSelectImages = (imgs) => {
+      setAttributes({
+        images: imgs,
+      })
+    }
+    const onChangeCaption = (newCaption, id) => {
+      const { images } = attributes;
+
+      for (let imgObj in images) {
```
Comment (vstepanyuk) [inline src/image_gallery/index.js:42]:
```js
images
  .filter(img => img.id === id)
  .forEach(img => img.caption = newCaption)
```

File: src/image_gallery/index.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      type: 'string',
+    }
+  },
+
+  edit({ attributes, className, setAttributes, isSelected }) {
+    // For debugging purpose within Drupal, please keep
+    console.log(attributes.images)
+
+    const hasImages = attributes.images && !!attributes.images.length;
+    const onSelectImages = (imgs) => {
```
Comment (vstepanyuk) [inline src/image_gallery/index.js:34]:
```js
const onSelectImages = (images) => setAttributes({ images })
```

File: src/image_gallery/index.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+            <MediaPlaceholder
+                addToGallery={hasImages}
+                isAppender={hasImages}
+                className={className}
+                disableMediaButtons={hasImages && !isSelected}
+                icon={!hasImages && 'dashicons-images-alt'}
+                labels={{
+                  title: !hasImages && __('Image Gallery'),
+                  instructions: !hasImages && 'Bitte Bilder auswählen',
```
Comment (vstepanyuk) [inline src/image_gallery/index.js:90]: what about translation here?

File: src/scripts.js [hunk 4]
```diff
@@ -1,6 +1,33 @@
...
-
-/*
-  If you use frontend scripts, please uncomment the
-  corresponding line on "webpack.config.js"
+/**
+ * Frontend script
+ * Note: This is only used for the image gallery custom block.
  */
+
+const gallery = document.getElementsByClassName('wp-block-coliquio-image-gallery');
```
Comment (vstepanyuk) [inline src/scripts.js:6]: document.querySelector?

Comment (col-tom) [reply src/scripts.js:6]: yep, works both

**Pull request merged**